include/exadg/time_integration/restart.h

```diff
@@ -138,6 +155,739 @@ read_write_distributed_vector(VectorType & vector, BoostArchiveType & archive)
+
+  JacobiPreconditioner<MassOperator<dim, n_components, Number>> jacobi_preconditioner(
+    mass_operator, true /* initialize_preconditioner */);
+
+  // Setup RemotePointEvaluation since the `source_vectors` all live on the same triangulation.
+  typename dealii::Utilities::MPI::RemotePointEvaluation<dim>::AdditionalData rpe_data(
+    rpe_tolerance_unit_cell,
+    rpe_enforce_unique_mapping,
+    0 /* rtree_level */,
+    {} /* marked_vertices */);
+
+  dealii::Utilities::MPI::RemotePointEvaluation<dim> rpe_source(rpe_data);
+
+  // The sequence of integration points follows from the sequence of points as encountered during
+  // cell batch loop.
+  std::vector<dealii::Point<dim>> integration_points_target =
+    collect_integration_points<dim, n_components, Number>(target_matrix_free,
+                                                          dof_index,
+                                                          quad_index);
+
+  rpe_source.reinit(integration_points_target,
+                    source_dof_handler.get_triangulation(),
+                    *source_mapping);
+  AssertThrow(rpe_source.all_points_found(),
+              dealii::ExcMessage("Could not interpolate source grid vector in target grid."));
+
+  CellIntegrator<dim, n_components, Number> fe_eval(target_matrix_free, dof_index, quad_index);
+
+  // Loop over vectors and project.
+  for(unsigned int i = 0; i < target_vectors.size(); ++i)
+  {
+    // Evaluate the source vector at the target integration points.
+    VectorType const & source_vector = *source_vectors.at(i);
+    source_vector.update_ghost_values();
+
+    std::vector<
+      typename dealii::FEPointEvaluation<n_components, dim, dim, Number>::value_type> const
+      values_source_in_q_points_target = dealii::VectorTools::point_values<n_components>(
+        rpe_source, source_dof_handler, source_vector, dealii::VectorTools::EvaluationFlags::avg);
+
+    // Assemble right hand side vector for the projection.
+    VectorType system_rhs = assemble_projection_rhs<dim, n_components, Number, VectorType>(
+      target_matrix_free, fe_eval, values_source_in_q_points_target, dof_index);
+
+    // CG solver for global projection.
+    unsigned int constexpr max_iter = 10000;
+    double const abs_tol            = 1e-16 * system_rhs.l2_norm();
+    double constexpr rel_tol        = 1e-12;
+
+    dealii::ReductionControl     reduction_control(max_iter, abs_tol, rel_tol);
+    dealii::SolverCG<VectorType> solver_cg(reduction_control);
+
+    VectorType sol;
+    sol.reinit(system_rhs, false /* omit_zeroing_entries */);
+
+    solver_cg.solve(mass_operator, sol, system_rhs, jacobi_preconditioner);
+
+    *target_vectors[i] = sol;
+
+    if(dealii::Utilities::MPI::this_mpi_process(target_dof_handler.get_communicator()) == 0)
+    {
+      std::cout << "    global projection required " << reduction_control.last_step()
+                << " CG iterations.\n";
+    }
+  }
```

Comment:
Please use ExaDG utility functions like `print_solver_info_linear()` for this (making `pcout` an argument of this function). See usage e.g. in https://github.com/exadg/exadg/blob/master/include/exadg/convection_diffusion/time_integration/time_int_bdf.cpp